src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -365,9 +387,12 @@ namespace AppInstaller::CLI::Workflow
             Workflow::ShowInstallationDisclaimer <<
             Workflow::ReportExecutionStage(ExecutionStage::Download) <<
             Workflow::DownloadInstaller <<
+            Workflow::ReportExecutionStage(ExecutionStage::PreExecution) <<
+            Workflow::SnapshotARPEntries <<

```

Comment:
Do we want to report these for upgrade scenarios?

Reply:
I have added it to upgrade.  In doing so, I coalesced all of the disparate install flows into a single task that should hopefully be the only one used to do actual download and installation from now on.